### DISCUSSION TITLE
Assigning multiple sublayers into a single trait

### DISCUSSION THREAD

**Discussion by soulmv** [nftchef/art-engine#28] [Help + Q&A] [answered]
Hi everyone,
I have a root folder "Hair" with 2 sublayers: 

"common#75"
 "rare#25"

There are weighted items in each sublayer. Everytime I generate json , it is showing something like " trait : Hair , value: common "
Is there any way to assign all these 2 sublayers to a root folder's single trait , so only file names in sublayers will be included in json?

Thanks a bunch!

**Comment by nftchef** [comment, marked as answer]
yes, by default sublayer traits inherit the name from the parent folder name. To turn this off, change the following to `false` in config.js

`const useRootTraitType = false;`

![image](https://user-images.githubusercontent.com/91582112/153914957-f8396939-116f-41e0-94ea-b037c243bd2d.png)

**Reply by soulmv** [reply]
Thanks for help and all your work! 

I actually figured it out without turning root trait option off : 
Even having SublayerOption configured to give subfolders particular trait names, I wanted to have these subfolders as trait names and files inside them as values (like if all these files from different folders were in one  root folder),  but was constantly getting root folder as trait name and subfolder name as a trait value. The problem was some clean name files contained in subfolders. Once I gave these files chance weight - problem was solved (didn't even need Sublayer Option to be configured anymore). Only logical explanation I see is that those files were selected first in the stack, thus altering trait value.

**Reply by nftchef** [reply]
yes, it's designed to use the subfolder name if it is weighted and includes unweighted (required) files.

The use case for that is if a single attribute contains multiple art layers, e.g. line and fill, or one part of hair on top, and one behind.
